### PR TITLE
UP 92 - Person Queries

### PR DESCRIPTION
**Pull request by Lou3797**
API page for Person object queries
_/api/person_

Queries in Viewer used:
- person(id)
- allPeople

There are some modified paths in various files due to some Windows shenanigans I ran into. Currently, Agent (capital A) is the correct folder path, so all instances of 'Agent' instead of 'agent' should be correct.

### CODE REVIEW

File: packages/app/pages/Api/person.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * Maps the array of Person to their individual pieces
+ */
+export const GetAllPeople = getAllPeople( ({ people, loading, error}) => {
+
+  if (loading) {
+    return <p>Loading...</p>
+  } else if (error) {
+    return (<p>Error!</p>) // This can be made more specific
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/person.tsx]: What's the difference between loading and error where error needs parentheses?

File: packages/app/pages/Api/person.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * Takes the given Person and returns its individual information
+ */
+export const GetSinglePerson = getPersonById( ({ person, loading, error }) => {
+
+  if (loading) {
+    return <p>Loading...</p>
+  } else if (error) {
+    return (<p>Error!</p>) // This can be made more specific
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/person.tsx]: Same question as above, what is the difference between loading and error?

File: packages/app/pages/Members/index.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 import * as React from 'react'
 import ProjectTemplate from '@vflows/views/templates/ProjectTemplate'
-import { AgentType } from '@vflows/bindings/agent/agent'
+import { AgentType } from '@vflows/bindings/Agent/agent'
```
Comment (ConnorHibbs) [inline packages/app/pages/Members/index.tsx]: (Not a comment specifically on this PR)
We should find a standard for naming folders. I encountered difficulties with Agent as well in my PRs where there becomes duplicate code when merging.

File: packages/app/routes.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ import ProcessesPage from "./pages/Processes"
 import InventoryPage from "./pages/Inventory"
```
Comment (ConnorHibbs) [inline packages/app/routes.tsx]: Please add semicolons to the import statements.

File: packages/ui-bindings/Person/getPersonById.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { connect } from "react-redux";
+import { gql, graphql, compose } from "react-apollo";
+
+import { AppState } from "@vflows/store/types.js";
+import { getActiveLoginToken } from "@vflows/store/selectors/auth.js";
+
+import { personInterface } from "./personInterface";
+
+const query = gql`
+query($token: String, $PersonId: Int) {
```
Comment (ConnorHibbs) [inline packages/ui-bindings/Person/getPersonById.tsx]: Naming variables with a capital letter (PersonId) can lead to confusion with objects.

**Review by ConnorHibbs — CHANGES_REQUESTED** [review]
Almost everything looks good! We can discuss the Agent / agent problem as a group, and then there are just a few small fixes

File: packages/ui-views/templates/ProjectTemplate/projectTemplate.tsx [hunk 6]
```diff
@@ -2,7 +2,7 @@ import * as React from 'react'
 import Sidebar from '../../organisms/Sidebar'
 import SecondaryMenu from '../../organisms/SecondaryMenu'
 import * as themeable from 'react-themeable'
-import BindAgent, { AgentType } from '@vflows/bindings/agent/agent'
+import BindAgent, { AgentType } from '@vflows/bindings/Agent/agent'
```
Comment (fontainesw) [inline packages/ui-views/templates/ProjectTemplate/projectTemplate.tsx]: these should be double quotes, and imports need semicolons

**Review by ConnorHibbs — APPROVED** [review]
After discussing Agent / agent, I'll approve this on the assumption that the mild linter errors are corrected.

File: packages/ui-views/organisms/Aside/aside.tsx [hunk 7]
```diff
@@ -3,7 +3,7 @@ import * as themeable from 'react-themeable'
 import { withRouter } from 'react-router'
 
 import BindOrgs from '@vflows/bindings/user/CurrentOrganizations'
-import { AgentType } from '@vflows/bindings/agent/agent'
+import { AgentType } from '@vflows/bindings/Agent/agent'
```
Comment (fontainesw) [inline packages/ui-views/organisms/Aside/aside.tsx]: see comment on project/projecttemplate

File: packages/ui-views/organisms/MembersBig/MembersBig.tsx [hunk 8]
```diff
@@ -1,8 +1,8 @@
 import * as React from 'react'
 import * as themeable from 'react-themeable'
 
-import BindMembers from '@vflows/bindings/agent/AgentMembers'
-import { AgentType } from '@vflows/bindings/agent/agent'
+import BindMembers from '@vflows/bindings/Agent/AgentMembers'
+import { AgentType } from '@vflows/bindings/Agent/agent'
```
Comment (fontainesw) [inline packages/ui-views/organisms/MembersBig/MembersBig.tsx]: see comment on project/projecttemplate

File: packages/app/pages/Api/person.tsx (hunk 1 above)
Comment (Lou3797) [reply packages/app/pages/Api/person.tsx]: There isn't a difference. I was experimenting with how to make errors more specific which led me to needing to add parenthesis to the return statement. I just never ended up removing them it seems when I reverted back to simple error messages.

File: packages/app/pages/Api/person.tsx (hunk 2 above)
Comment (Lou3797) [reply packages/app/pages/Api/person.tsx]: See comment above.

File: packages/app/pages/Accounts/accounts.tsx [hunk 9]
```diff
@@ -8,7 +8,7 @@
 
 import * as React from 'react'
 import { Component, ReactElement } from 'react'
-import { AgentType } from '@vflows/bindings/agent/agent'
+import { AgentType } from '@vflows/bindings/Agent/agent'
```
Comment (fontainesw) [inline packages/app/pages/Accounts/accounts.tsx]: single quotes should be double quotes

File: packages/app/pages/Members/index.tsx (hunk 3 above)
Comment (fontainesw) [reply packages/app/pages/Members/index.tsx]: this was changed to match the standards of Ui bindings

**Review by fontainesw — CHANGES_REQUESTED** [review]

File: packages/ui-views/templates/ProjectTemplate/projectTemplate.tsx (hunk 6 above)
Comment (Lou3797) [reply packages/ui-views/templates/ProjectTemplate/projectTemplate.tsx]: Noted. This change was automatically done by WebStorm so I didn't manually inspect each change. I'll go ahead and refactor these files.

**Review by Ryan-Guinn — APPROVED** [review]

**Review by fontainesw — APPROVED** [review]
Looks good

**Pull request merged**